### ISSUE TITLE
Import Email > HTML to Plain Text > Broken of cut off

### ISSUE THREAD

**Issue by bastiaanc** [mkoch227/Mods-for-HESK#87]
Hi,

If an email is imported into HESK > the text is converted in plain text if it wasn't already
My problem now is > That is comes in like this:

FREE Bowling – Single Airman Pre Labor Day Special!

All Single Airman are invited to enjoy free bowling and food specials
at the Ramstein Bowling Center on Friday, August 29th from 2000-2300.
All 30 lanes will be available, along with FREE shoe rentals! All you
have to do is “Show Up” and bowl. For more details, please call
0xxx-47-xxx.

--------------------------------------------------------------------------------------------------------------------------------------------

Ii would like to have it like this:

Have you signed up for the FSS eSpotlight newsletter yet? You could WIN a FSS gift card! See details below.

FREE Bowling – Single Airman Pre Labor Day Special!

All Single Airman are invited to enjoy free bowling and food specials at the Ramstein Bowling Center on Friday, August 29th from 2000-2300. All 30 lanes will be available, along with FREE shoe rentals! All you have to do is “Show Up” and bowl. For more details, please call xxx-47-xxx
Is this something that can be done?

--------------------------------------------------------------------------------------------------------------------------------------------

Please see > discussion in the http://developers.phpjunkyard.com Forum:
http://developers.phpjunkyard.com/viewtopic.php?f=13&t=5034&p=21977&hilit=broken+text#p21977

Thanks for all you do :-)

**Comment by bastiaanc** [comment]
Hi Mike,

Here is an example text.

All Single Airman are invited to enjoy free bowling and food specials at the Ramstein Bowling Center on Friday, August 29th from 2000-2300. All 30 lanes will be available, along with FREE shoe rentals! All you have to do is “Show Up” and bowl. For more details, please call xxx-47-xxx


As soon as the text is imported by email it looks a little like this:

All Single Airman are invited to enjoy
free bowling and food specials at the
Ramstein Bowling Center on
 Friday, August 29th
from 2000-2300. All 30 lanes
 will be available
along with FREE shoe rentals!
All you have to do is “Show Up” 
and bowl. For more details, please
 call xxx-47-xxx


I hope it is more clear now ;-)

Thanks :-)

**Comment by mkoch227** [comment]
Can you include the actual HTML markup of the email? I can't tell why line breaks are being added when I only see plain text

**Comment by bastiaanc** [comment]
Hi MikeŠ.
It is not text in teh email. It is text in the ticket.
So the text from the email. It does not matter what kind of text it is.
Only in teh imported ticket teh text is ³cut-off²

Van:  Mike Koch <notifications@github.com>
Beantwoorden - Aan:  mkoch227/Mods-for-HESK
<reply+008cacdfa4f66db5e51875d5f5936b8f83311fb0d368820b92cf0000000110bec2489
2a169ce0301d259@reply.github.com>
Datum:  vrijdag 2 januari 2015 21:32
Aan:  mkoch227/Mods-for-HESK <Mods-for-HESK@noreply.github.com>
CC:  Bas Cuppen <bas@cuppen.de>
Onderwerp:  Re: [Mods-for-HESK] Import Email > HTML to Plain Text > Broken
of cut off (#87)


Can you include the actual HTML markup of the email? I can't tell why line
breaks are being added when I only see plain text

‹
Reply to this email directly or view it on GitHub
<https://github.com/mkoch227/Mods-for-HESK/issues/87#issuecomment-68559670>
.

**Comment by mkoch227** [comment]
I understand that the text in the ticket is incorrect. However, I cannot determine as to why it's happening unless I can see a copy of the original e-mail.  HESK uses PHP's `nl2br` function to convert new lines (`\r\n`, `\n\r`, `\n` and `\r`) to HTML's line break tag (`<br>`). It's possible that the message contains one or more of those escape characters and thus is being converted to a line break tag.

**Comment by bastiaanc** [comment]
Can I send a ticket via an email address? Then you can see itŠ.

Van:  Mike Koch <notifications@github.com>
Beantwoorden - Aan:  mkoch227/Mods-for-HESK
<reply+008cacdfcdfc85a51a64da6a611ebca593f837c2a675194c92cf0000000110c154269
2a169ce0301d259@reply.github.com>
Datum:  zondag 4 januari 2015 20:19
Aan:  mkoch227/Mods-for-HESK <Mods-for-HESK@noreply.github.com>
CC:  Bas Cuppen <bas@cuppen.de>
Onderwerp:  Re: [Mods-for-HESK] Import Email > HTML to Plain Text > Broken
of cut off (#87)


I understand that the text in the ticket is incorrect. However, I cannot
determine as to why it's happening unless I can see a copy of the original
e-mail.  HESK uses PHP's nl2br function to convert new lines (\r\n, \n\r, \n
and \r) to HTML's line break tag (<br>). It's possible that the message
contains one or more of those escape characters and thus is being converted
to a line break tag.

‹
Reply to this email directly or view it on GitHub
<https://github.com/mkoch227/Mods-for-HESK/issues/87#issuecomment-68644949>
.

**Comment by mkoch227** [comment]
Send the email to hesk-test [at] mkochcs [dot]com

**Comment by bastiaanc** [comment]
Email is out :-)

Van:  Mike Koch <notifications@github.com>
Beantwoorden - Aan:  mkoch227/Mods-for-HESK
<reply+008cacdfe4bc20a8158e358ab2eb7f0f4a890c335e5b1fe992cf0000000110c15aac9
2a169ce0301d259@reply.github.com>
Datum:  zondag 4 januari 2015 20:46
Aan:  mkoch227/Mods-for-HESK <Mods-for-HESK@noreply.github.com>
CC:  Bas Cuppen <bas@cuppen.de>
Onderwerp:  Re: [Mods-for-HESK] Import Email > HTML to Plain Text > Broken
of cut off (#87)


Send the email to hesk-test [at] mkochcs [dot]com

‹
Reply to this email directly or view it on GitHub
<https://github.com/mkoch227/Mods-for-HESK/issues/87#issuecomment-68646033>
.

**Comment by mkoch227** [comment]
Ok, so the same issue occurred. I did some research, and there appears to be a reason, which is explained at https://mathiasbynens.be/notes/gmail-plain-text. Because the issue is on the mail server that is retrieving the emails, HESK and Mods for HESK will be unable to determine which line breaks are actual line breaks, and which line breaks are due to the 78 character limit. You'll want to make changes to your mail server to render plaintext emails to use one of the methods provided in the article I linked.
